uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-popup-details-tab.component.spec.js

```diff
@@ -919,21 +919,54 @@ define([
             });
 
             describe('when set times are outside the allowed range', function () {
-              var minTime = '09:00';
-              var maxTime = '17:00';
+              initSetTimeAccordingToWorkPattern('02:00', '23:00', '09:00', '17:00');
 
+              it('sets from time to minimum allowed time', function () {
+                expect(controller.uiOptions.times.from.time).toBe('09:00');
+                expect(leaveRequest.from_date.slice(11, 16)).toBe('09:00');
+              });
+
+              it('sets to time to maximum allowed time', function () {
+                expect(controller.uiOptions.times.to.time).toBe('17:00');
+                expect(leaveRequest.to_date.slice(11, 16)).toBe('17:00');
+              });
+            });
+
+            describe('when work pattern day has change to a non-working day', function () {
+              initSetTimeAccordingToWorkPattern('02:00', '23:00', '', '');
+
+              it('sets from time to minimum allowed time', function () {
+                expect(controller.uiOptions.times.from.time).toBe('00:00');
+                expect(leaveRequest.from_date.slice(11, 16)).toBe('00:00');
+              });
+
+              it('sets to time to maximum allowed time', function () {
+                expect(controller.uiOptions.times.to.time).toBe('00:00');
+                expect(leaveRequest.to_date.slice(11, 16)).toBe('00:00');
+              });
+            });
+
+            /**
+             * Initialises set time for a given work pattern range
+             *
+             * @param {String} setFromTime HH:mm
+             * @param {String} setToTime HH:mm
+             * @param {String} wpFromTime HH:mm or empty string
```

Comment:
Nobody will lose an extra second of their time if you use `workPatternFromTime` and `workPatternToTime` instead.

But, if you have a function that is called `setupWorkPatternTimeResponse` (or similar) you can forget about the `workPattern` prefix and just use `fromTime` and `toTime`.

Reply:
Fixed as per previous comment